src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Usage/CSharpImplementGenericMathInterfacesCorrectly.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Usage;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Usage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpImplementGenericMathInterfacesCorrectly : ImplementGenericMathInterfacesCorrectly
+    {
+        protected override SyntaxNode? FindTheTypeArgumentOfTheInterfaceFromTypeDeclaration(ISymbol typeSymbol, ISymbol anInterfaceSymbol)
+        {
+            foreach (SyntaxReference syntaxReference in typeSymbol.DeclaringSyntaxReferences)
+            {
+                SyntaxNode typeDefinition = syntaxReference.GetSyntax();
+                if (typeDefinition is ClassDeclarationSyntax classDeclaration)
```

Comment:
What about interfaces. For example:

`interface IMyInterface : IAdditionOperators<int, int, int> { }`

Reply:
Are `records` covered under `ClassDeclarationSyntax` and `StructDeclarationSyntax` or are they separate?

Reply:
Did nor think we need to check interfaces (thought need to check only interface implementation 😄), but yes makes sense to check interfaces